### ISSUE TITLE
styled jsx only works client side when using custom _app.js

### ISSUE THREAD

**Issue by shaneosullivan** [zeit/next.js#8868]
# Bug report
When using a custom _app.js file, styled jsx content is only added when running in the browser

## Describe the bug
When loading a page, it flickers as the styled jsx CSS is not loaded on the server side, and is applied when the JavaScript loads.

## To Reproduce

Steps to reproduce the behavior, please provide code snippets or a repository:

Code example:

```
import React from "react";
import App, { Container } from "next/app";

class MyApp extends App {
  render() {
    const { Component, pageProps } = this.props;

    return (
      <Container>
        <Component {...pageProps} />
      </Container>
    );
  }
}

MyApp.getInitialProps = async (context: any) => {
  let props: any = await App.getInitialProps(context);

  return { ...props };
};

export default MyApp;
```
You can see this happening on our site at promise-pay.com, just disable JavaScript to see the difference.

The page props being generated by `App` are below, with no mention of styles.
```
{ pageProps: { host: 'localhost:3000', namespacesRequired: [ 'common' ] } }
```

## Expected behavior

The styles should be rendered both server side and client side

## Screenshots

![Screen Shot 2019-09-26 at 10 25 54 AM](https://user-images.githubusercontent.com/49378/65710554-1371c100-e048-11e9-9786-128058cf1811.png)
![Screen Shot 2019-09-26 at 10 25 34 AM](https://user-images.githubusercontent.com/49378/65710555-1371c100-e048-11e9-9488-aa42e7e13525.png)

## System information

- OS: macOS
- Browser: Chrome
- Version of Next.js: 
```
"@zeit/next-css": "^1.0.1",
"@zeit/next-less": "^1.0.1",
"next": "^8.1.0",
```

## Additional context

This is a similar effect to that seen when using a custom _document.js as documented in https://github.com/zeit/next.js/issues/2593, but the proposed solution, to use the parent `getInitialProps` doesn't work.

**Comment by shaneosullivan** [comment]
It turns out that the `_app.ts` file was not the issue. Our app's `package.json` had a dependency on `styled-jsx`, and removing this fixed the issue.

The investigation into the `node_modules` execution showed that `styled-jsx` has a singleton for storing the css that each component creates, and a `flush()` function to fetch it.  Because of our root level dependency on `styled-jsx`, our components were registering their css with one copy of the singleton, but Next was calling `flush()` on a different copy of the singleton, thereby getting back an empty array of styles.  This is more for historical context for anyone else running into this, I'm unblocked right now

**Comment by balazsorban44** [comment]
This issue has been automatically locked due to no recent activity. If you are running into a similar issue, please create a new issue with the steps to reproduce. Thank you.
